app/javascript/history/history.jsx

```diff
@@ -0,0 +1,212 @@
+import { h, Component } from 'preact';
+import { PropTypes } from 'preact-compat';
+import debounce from 'lodash.debounce';
```

Comment:
I'm using `lodash.debounce` to avoid sending every keystroke to Algolia. It's configured to debounce up to 300ms